### PR TITLE
lmstudio: 0.3.3 -> 0.3.4

### PR DESCRIPTION
**Pull request by crertel**
## Changes

[Full Changelog](https://lmstudio.ai/blog/lmstudio-v0.3.3)

### Summary

What’s new in 0.3.4:

**New**
* MLX support
* MLX KV caching
* Mission Control: Cmd+Shift+M to search for models, Cmd+Shift+R to manage LM Runtimes
* Set Structured Output (JSON Schema) from the UI

**Bug Fixes**

  * Fix for Black Screen after prolonged use (reference: [lmstudio-bug-tracker#98](https://github.com/lmstudio-ai/lmstudio-bug-tracker/issues/98))
  * Fix for no port other than 1234 working for the local server (reference: [lms#80](https://github.com/lmstudio-ai/lms/issues/80))
  * Fix for embedding API not working from Obsidian (reference: [tracker#142](https://github.com/lmstudio-ai/lmstudio-bug-tracker/issues/142))
  * Fix for RAG sometimes failing during document processing
    

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by eeedean** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 347599`

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>lmstudio</li>
  </ul>
</details>

### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>lmstudio</li>
  </ul>
</details>

---

Did they change the URL for aarch64-darwin? 
it seems to be now:

https://releases.lmstudio.ai/darwin/arm64/0.3.4/3/LM-Studio-0.3.4-arm64.dmg

instead of 

https://releases.lmstudio.ai/darwin/arm64/0.3.4/LM-Studio-0.3.4-arm64.dmg

They finally added the build revision into the URL for aarch64-darwin, too. Did you test it beforehand? I suspect, it did not ever work with the current URL in the PR.

File: pkgs/by-name/lm/lmstudio/darwin.nix [hunk 1]
```diff
@@ -10,7 +10,7 @@ stdenv.mkDerivation {
 
   src = fetchurl {
     url = "https://releases.lmstudio.ai/darwin/arm64/${version}/LM-Studio-${version}-arm64.dmg";
```
Comment (eeedean) [inline pkgs/by-name/lm/lmstudio/darwin.nix:12]: - [ ] url needs change including build revision:

https://releases.lmstudio.ai/darwin/arm64/${version}/3/LM-Studio-${version}-arm64.dmg

maybe, we even want to put the build revision into its own variable, as it is present in both URLs now?
https://releases.lmstudio.ai/darwin/arm64/${version}/${build-revision}/LM-Studio-${version}-arm64.dmg

**Review by eeedean — CHANGES_REQUESTED** [review]

File: pkgs/by-name/lm/lmstudio/darwin.nix (hunk 1 above)
Comment (crertel) [reply pkgs/by-name/lm/lmstudio/darwin.nix:12]: Perhaps--but the thing is, they keep changing around the URLs on us (at least for the Linux folks--Mac seems stabler). I'm kinda wondering if we should go the other way and just hardcode the URL every time until they get their act together.

**Comment by crertel** [comment]
@eeedean thanks for catching that bad URL, must've fat-fingered it. Not on Mac so my testing there isn't super thorough.

**Review by eeedean — CHANGES_REQUESTED** [review]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 347599`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>lmstudio</li>
  </ul>
</details>

### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>lmstudio</li>
  </ul>
</details>

- [ ] I would appreciate a squash of the commits, please. Then it's perfect imo. 🐱 

Edit: Also sorry for taking so long to get back to you ☹️

**Comment by crertel** [comment]
@eeedean done!

**Review by eeedean — APPROVED** [review]

**Comment by eeedean** [comment]
In case, you don’t know this friend, yet: https://github.com/nixos/nixpkgs-merge-bot#features

**Comment by crertel** [comment]
@NixOS/nixpkgs-merge-bot merge

**Comment by eeedean** [comment]
I could swear, I already used this bad boy to merge one of my PRs… I‘ll investigate again 😅

**Comment by eeedean** [comment]
@crertel , merge bot seems to be an in-progress-RFC and not sharp yep. Currently, it does only mention the merge queue issue, as seen. That does at least motivate some commiter (in this case Lassalus, who also is involved in creation of the bot and RFC — Thanks!!) to press the holy button.

See for more information [here](https://github.com/NixOS/rfcs/pull/172). 

Anyway: seems reasonable to let the bot put an approved PR into the merge queue. Works as designed!

**Pull request merged**